geode-experimental-driver/src/test/java/org/apache/geode/experimental/driver/RegionIntegrationTest.java

```diff
@@ -94,6 +94,29 @@ public void putWithJSONKeyAndValue() throws Exception {
     assertTrue(PdxInstance.class.isAssignableFrom(entry.getValue().getClass()));
   }
 
+  @Test
+  public void putIfAbsent() throws Exception {
+    Region<JSONWrapper, JSONWrapper> region = driver.getRegion("region");
+    JSONWrapper document = JSONWrapper.wrapJSON(jsonDocument);
+
+    assertNull(region.putIfAbsent(document, document));
+
+    JSONWrapper value = region.get(document);
+    assertEquals(document, value);
+    assertEquals(1, serverRegion.size());
+
+    assertEquals(document, region.putIfAbsent(document, JSONWrapper.wrapJSON("{3 : 2}")));
+    value = region.get(document);
+    assertEquals(document, value);
+    assertEquals(1, serverRegion.size());
+
+    org.apache.geode.cache.Region.Entry entry =
+        (org.apache.geode.cache.Region.Entry) serverRegion.entrySet().iterator().next();
+
+    assertTrue(PdxInstance.class.isAssignableFrom(entry.getKey().getClass()));
+    assertTrue(PdxInstance.class.isAssignableFrom(entry.getValue().getClass()));
+  }
```

Comment:
Add a test for putIfAbsent(k,null) and assert that the region holds an entry that is INVALID.

Reply:
I don't think we need that integration test. That's a matter of what `PutIfAbsent` does; the tests for the operation handler just need to make sure that it's hooked up properly. I'll add a unit test for nulls.